src/client/Microsoft.Identity.Client/Instance/Discovery/RegionDiscoveryProvider.cs

```diff
@@ -27,11 +28,7 @@ public async Task<InstanceDiscoveryMetadataEntry> GetMetadataAsync(Uri authority
                 return null;
             }
 
-            // already regionalized
-            if (authority.Host.StartsWith(region))
-            {
-                return CreateEntry(requestContext.ServiceBundle.Config.AuthorityInfo.Host, authority.Host);
-            }
+            
```

Comment:
We need this. Otherwise `GetRegionalizedEnviroment` will end up regionalizing regionalized URL like so region.region.domain. I think the `SuccessfulResponseFromRegionalizedAuthorityAsync` unit test covers this scenario.

Reply:
#Resolved